src/main/java/com/vinted/flink/bigquery/sink/defaultStream/BigQueryDefaultSinkWriter.java

```diff
@@ -107,32 +97,70 @@ public AppendCallBack(BigQueryDefaultSinkWriter<A, ?> parent, String traceId, Ro
         }
 
         @Override
-        public void onFailure(Throwable t) {
-            logger.info("Trace-id {} Received error {}", t.getMessage(), traceId);
+        public void onSuccess(AppendRowsResponse result) {
+            Optional.ofNullable(this.parent.metrics.get(rows.getStream())).ifPresent(m -> {
+                m.setBatchCount(rows.getData().size());
+                m.setOffset(result.getAppendResult().getOffset().getValue());
+            });
             this.parent.inflightRequestCount.arriveAndDeregister();
+        }
+
+        private void retryWrite(Throwable t, int newRetryCount) {
```

Comment:
Nitpicking: but shouldn't we try to move private functions closer to the end of the file?